File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.py
def solution(n, wires):
    ans=100
    arr = []
    wires.sort()
    for i in range(len(wires)):
        w = wires[:i] + wires[i+1:]
        arr.append(w[0][0])
        arr.append(w[0][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
        for j in range(1,len(w)):
            if w[j][0] not in arr and w[j][1] in arr:
                arr.append(w[j][0])
            if w[j][1] not in arr and w[j][0] in arr:
                arr.append(w[j][1])
                
                
        if abs(len(arr)-(n-len(arr)))<ans:
            ans = abs(len(arr)-(n-len(arr)))
        arr = []
    return ans

File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.js
function solution(n, wires) {
    let min = Infinity
    const bfs = (x,graph) => {
        let visited=Array(n).fill(false)
        visited[x-1]=true
        let queue = [x]
        while(queue.length>0){
            let curr = queue.shift()
            for(let next of graph[curr]){
                if(visited[next-1]) continue
                queue.push(next)
                visited[next-1]=true
            }
        }
        let t=0
        let f=0
        for(let i=0;i<visited.length;i++){
            if(visited[i]) t++
            else f++
        }
        return Math.abs(t-f)
    }
    
    for(let i=0;i<n-1;i++){
        let graph = {}
        let visited = Array(n).fill(false)
        for(let j=0;j<n-1;j++){
            if(i===j) continue
            if(!graph[wires[j][0]]) graph[wires[j][0]] = []
            if(!graph[wires[j][1]]) graph[wires[j][1]] = []
            graph[wires[j][0]].push(wires[j][1])
            graph[wires[j][1]].push(wires[j][0])
        }
        let num
        if(graph[1]) num=bfs(1,graph)
        else num=bfs(2,graph)
        min=Math.min(num,min)
    }
    return min
}